### PR TITLE
Upload Integration Test Binary To S3 Instead Of Building On EC2 Fixes…

### PR DESCRIPTION
**Pull request by sethAmazon**
… (#364)

# Description of the issue
Reduce time it takes to run the integration tests by only building once. 

# Description of changes
Upload binary to s3 instead of build on ec2

# Tests
Ran the integration tests on my fork

### CODE REVIEW

File: integration/terraform/ec2/linux/variables.tf [hunk 1]
```diff
@@ -76,4 +76,8 @@ variable "ca_cert_path" {
 variable "arc" {
   type = string
   default = ""
+}
+
+variable "binary_name" {
```
[the remark this answers is not in the document]
Comment (sethAmazon) [reply integration/terraform/ec2/linux/variables.tf:81]: Not really but I should

**Comment by khanhntd** [comment]
Why you are at it, please [delete these two variables](https://github.com/aws/amazon-cloudwatch-agent/pull/365/files#diff-9b9f9a38b889c9dcb55ba0330baa69c8fdd62ac1100b6eb59b6fc44008f5d320L54-L55)  and replace it with `aws-actions/configure-aws-credentials@v1`

File: .github/workflows/integrationTest.yml [hunk 2]
```diff
@@ -15,7 +15,60 @@ concurrency:
...
 jobs:
+  MakeBinary:
+    name: 'MakeBinary'
+    runs-on: ubuntu-latest
+    steps:
+      # Set up building environment, patch the dev repo code on dispatch events.
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15
```
[the remark this answers is not in the document]
Comment (khanhntd) [reply .github/workflows/integrationTest.yml:26]: I would prefer the other way around since it would use the latest patch from go minor 1.15.

**Comment by sethAmazon** [comment]
I'll close this pr and just add this as a part of an effort to add more ami we support for testing

**Pull request closed without merging**